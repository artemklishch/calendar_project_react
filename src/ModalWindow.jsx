import React from 'react';

const ModalWindow = ({isOpen, onHideForm, tempObj}) => {
  if(!isOpen) return null;
  return (
    <div className="popup-layer">
      <form className="popup event" data-id-event="">
        <span className="popup__btn-close" onClick={onHideForm}><img className="close" src="https://img.icons8.com/color/48/000000/close-window.png" alt="close" /></span>
        <input className="event__name" name="header" type="text" placeholder="Add title " />
        <div className="popup__picker">
          <img className="clock" src="https://img.icons8.com/pastel-glyph/64/000000/time.png" alt="clock" />
          <input className="event__date-start" name="startTime" required type="date" defaultValue={tempObj.date} />
          <input type="time" className="startTime_place" name="startTimePlace" defaultValue={tempObj.startTime} />
          <span className='line'></span>
          <input type="time" className="endTime_place" name="endTimePlace" defaultValue={tempObj.endTime} />
          <input className="event__date-end" name="endTime" required type="date" defaultValue={tempObj.date} />
        </div>
        <div className="centerData">
          <img className="multiline" src="https://img.icons8.com/windows/32/000000/multiline-text.png" alt="mutiline" />
          <textarea type="text" className="multiline__text" placeholder="add description" name="description"></textarea>
          <input type="color" className="pick_color" name="backgroundColor" />
        </div>
        <div className="footer-popup">
          <p className="message_validation"></p>
          <div className="buttons">
            <img className="event__btn-delete" src="https://img.icons8.com/windows/32/000000/trash.png" alt="basket" />
            <button type="submit" className="event__btn-save btn_save">Save</button>
          </div>
        </div>
      </form>
    </div>
  );
};
export default ModalWindow;