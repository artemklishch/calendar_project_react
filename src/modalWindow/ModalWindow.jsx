import React from 'react';
import './popup.scss';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const ModalWindow = ({
  isOpen,
  onHideForm,
  tempObj,
  onCreateEvent,
  backgroundColor,
  isEditing,
  onDeleteEvent,
  onValidate,
  validateText,
  onLateEditing }) => {

  if (!isOpen) return null;

  const basketDeleteClass = classNames('event__btn-delete',
    { 'btn-delete_block': isEditing });
    
  return (
    <div className="popup-layer">
      <form className="popup event" onSubmit={onCreateEvent} >
        <span className="popup__btn-close" onClick={onHideForm}><img className="close" src="https://img.icons8.com/color/48/000000/close-window.png" alt="close" /></span>
        <input className="event__name" name="header" type="text" placeholder="Add title " defaultValue={tempObj.header} disabled={onLateEditing} />
        <div className="popup__picker" onInput={onValidate}>
          <img className="clock" src="https://img.icons8.com/pastel-glyph/64/000000/time.png" alt="clock" />
          <input className="event__date-start input" name="startDate" required type="date" defaultValue={tempObj.startDate} disabled={onLateEditing} />
          <input type="time" className="startTime_place input" name="startTimePlace" defaultValue={tempObj.startTime} disabled={onLateEditing} />
          <span className='line'></span>
          <input type="time" className="endTime_place input" name="endTimePlace" defaultValue={tempObj.endTime} disabled={onLateEditing} />
          <input className="event__date-end input" name="endDate" required type="date" defaultValue={tempObj.endDate} disabled={onLateEditing} />
        </div>
        <div className="centerData">
          <img className="multiline" src="https://img.icons8.com/windows/32/000000/multiline-text.png" alt="mutiline" />
          <textarea type="text" className="multiline__text" placeholder="add description" name="description" defaultValue={tempObj.description} disabled={onLateEditing}></textarea>
          <input type="color" className="pick_color" name="backgroundColor" defaultValue={backgroundColor} />
        </div>
        <div className="footer-popup">
          <p className="message_validation">{validateText}</p>
          <div className="buttons">
            <img id='delete-img' className={basketDeleteClass} src="https://img.icons8.com/windows/32/000000/trash.png" alt="basket" onClick={onDeleteEvent} />
            <button type="submit" className="event__btn-save btn_save">Save</button>
          </div>
        </div>
      </form>
    </div>
  );
};
export default ModalWindow;


ModalWindow.propTypes = {
  isOpen: PropTypes.bool,
  onHideForm: PropTypes.func.isRequired,
  onDeleteEvent: PropTypes.func.isRequired,
  onValidate: PropTypes.func.isRequired,
  validateText: PropTypes.string,
  backgroundColor: PropTypes.string,
}
ModalWindow.defaultProps = {
  validateText: '',
  backgroundColor: '#0851f6',
}