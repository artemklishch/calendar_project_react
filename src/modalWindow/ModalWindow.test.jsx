import React from 'react';
import { shallow } from 'enzyme';
import ModalWindow from './ModalWindow';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import toJson from 'enzyme-to-json';

describe('ModalWindow', () => {
  it('should display ModalWindow', () => {
    const tempObj = {
      header: 'My header',
      startDate: '2020-01-01',
      startTime: '10-00',
      endDate: '2020-01-01',
      endTime: '12-00',
      description: 'text',
    };
    const props = {
      isOpen: true,
      onHideForm: jest.fn(),
      onDeleteEvent: jest.fn(),
      onValidate: jest.fn(),
      validateText: 'text 1',
      backgroundColor: 'red',
      tempObj,
      onCreateEvent: jest.fn(),
      onLateEditing: false,
    };
    const wrappedComponent = shallow(<ModalWindow {...props} />);
    expect(toJson(wrappedComponent)).toMatchSnapshot();
  })

  it('should call function on submit', () => {
    const props = {
      isOpen: true,
      onHideForm: jest.fn(),
      onDeleteEvent: jest.fn(),
      onValidate: jest.fn(),
      validateText: 'text 1',
      backgroundColor: 'red',
      tempObj: {},
      onCreateEvent: jest.fn(),
      onLateEditing: false,
    };
    const wrappedComponent = shallow(<ModalWindow {...props} />);
    wrappedComponent.find('.popup').simulate('submit');
    expect(props.onCreateEvent).toBeCalled();
  })

  it('should call function on close', () => {
    const props = {
      isOpen: true,
      onHideForm: jest.fn(),
      onDeleteEvent: jest.fn(),
      onValidate: jest.fn(),
      validateText: 'text 1',
      backgroundColor: 'red',
      tempObj: {},
      onCreateEvent: jest.fn(),
      onLateEditing: false,
    };
    const wrappedComponent = shallow(<ModalWindow {...props} />);
    wrappedComponent.find('.popup__btn-close').simulate('click');
    expect(props.onHideForm).toBeCalled();
  })


  it('should call function on delete', () => {
    const props = {
      isOpen: true,
      onHideForm: jest.fn(),
      onDeleteEvent: jest.fn(),
      onValidate: jest.fn(),
      validateText: 'text 1',
      backgroundColor: 'red',
      tempObj: {},
      onCreateEvent: jest.fn(),
      onLateEditing: false,
    };
    const wrappedComponent = shallow(<ModalWindow {...props} />);
    wrappedComponent.find('#delete-img').simulate('click');
    expect(props.onDeleteEvent).toBeCalled();
  })
});