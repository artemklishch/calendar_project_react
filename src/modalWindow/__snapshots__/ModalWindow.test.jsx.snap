// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWindow should display ModalWindow 1`] = `
<div
  className="popup-layer"
>
  <form
    className="popup event"
    onSubmit={[MockFunction]}
  >
    <span
      className="popup__btn-close"
      onClick={[MockFunction]}
    >
      <img
        alt="close"
        className="close"
        src="https://img.icons8.com/color/48/000000/close-window.png"
      />
    </span>
    <input
      className="event__name"
      defaultValue="My header"
      disabled={false}
      name="header"
      placeholder="Add title "
      type="text"
    />
    <div
      className="popup__picker"
      onInput={[MockFunction]}
    >
      <img
        alt="clock"
        className="clock"
        src="https://img.icons8.com/pastel-glyph/64/000000/time.png"
      />
      <input
        className="event__date-start input"
        defaultValue="2020-01-01"
        disabled={false}
        name="startDate"
        required={true}
        type="date"
      />
      <input
        className="startTime_place input"
        defaultValue="10-00"
        disabled={false}
        name="startTimePlace"
        type="time"
      />
      <span
        className="line"
      />
      <input
        className="endTime_place input"
        defaultValue="12-00"
        disabled={false}
        name="endTimePlace"
        type="time"
      />
      <input
        className="event__date-end input"
        defaultValue="2020-01-01"
        disabled={false}
        name="endDate"
        required={true}
        type="date"
      />
    </div>
    <div
      className="centerData"
    >
      <img
        alt="mutiline"
        className="multiline"
        src="https://img.icons8.com/windows/32/000000/multiline-text.png"
      />
      <textarea
        className="multiline__text"
        defaultValue="text"
        disabled={false}
        name="description"
        placeholder="add description"
        type="text"
      />
      <input
        className="pick_color"
        defaultValue="red"
        name="backgroundColor"
        type="color"
      />
    </div>
    <div
      className="footer-popup"
    >
      <p
        className="message_validation"
      >
        text 1
      </p>
      <div
        className="buttons"
      >
        <img
          alt="basket"
          className="event__btn-delete"
          id="delete-img"
          onClick={[MockFunction]}
          src="https://img.icons8.com/windows/32/000000/trash.png"
        />
        <button
          className="event__btn-save btn_save"
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</div>
`;
